### PR TITLE
V6: cookie authentication without Identity. /8

### LINKED ISSUE #24515: .NET 6: Use cookie authentication without ASP.NET Core Identity
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d2229fc6-f8c8-952e-e567-b1d5a463055a
* Version Independent ID: b18a015b-914d-f014-f711-120d208904d9
* Content: [Use cookie authentication without ASP.NET Core Identity](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0)
* Content Source: [aspnetcore/security/authentication/cookie.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/cookie.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #24515
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0&branch=pr-en-us-24516)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@serpent5  can you take a quick look and compare to the original. Mostly just updated to .NET 6.

File: aspnetcore/security/authentication/cookie.md [hunk 1]
```diff
@@ -4,53 +4,45 @@ author: rick-anderson
 description: Learn how to use cookie authentication without ASP.NET Core Identity.
 monikerRange: '>= aspnetcore-2.1'
```
Comment (serpent5) [inline aspnetcore/security/authentication/cookie.md:5]:
```suggestion
monikerRange: '>= aspnetcore-3.1'
```

File: aspnetcore/security/authentication/cookie.md [hunk 2]
```diff
@@ -4,53 +4,45 @@ author: rick-anderson
...
+## Add cookie authentication
 
-In the `Startup.ConfigureServices` method, create the Authentication Middleware services with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication*> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie*> methods:
+* Add the Authentication Middleware services with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication*> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie*> methods.
+* Call [`UseAuthentication`](xref:Microsoft.AspNetCore.Builder.AuthAppBuilderExtensions.UseAuthentication%2A) and [`UseAuthorization`](xref:Microsoft.AspNetCore.Builder.AuthorizationAppBuilderExtensions.UseAuthorization%2A) to set the `HttpContext.User` property and run Authorization Middleware for requests. `UseAuthentication` and `UseAuthorization` must be called before `Map` methods such as <xref:Microsoft.AspNetCore.Builder.RazorPagesEndpointRouteBuilderExtensions.MapRazorPages%2A> and <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapDefaultControllerRoute%2A>
```
Comment (serpent5) [inline aspnetcore/security/authentication/cookie.md:26]:
```suggestion
* Add the Authentication Middleware services with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication%2A> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie%2A> methods.
* Call <xref:Microsoft.AspNetCore.Builder.AuthAppBuilderExtensions.UseAuthentication%2A> and <xref:Microsoft.AspNetCore.Builder.AuthorizationAppBuilderExtensions.UseAuthorization%2A> to set the `HttpContext.User` property and run the Authorization Middleware for requests. `UseAuthentication` and `UseAuthorization` must be called before `Map` methods such as <xref:Microsoft.AspNetCore.Builder.RazorPagesEndpointRouteBuilderExtensions.MapRazorPages%2A> and <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapDefaultControllerRoute%2A>
```

File: aspnetcore/security/authentication/cookie.md [hunk 3]
```diff
@@ -4,53 +4,45 @@ author: rick-anderson
...
 The authentication cookie's <xref:Microsoft.AspNetCore.Http.CookieBuilder.IsEssential> property is set to `true` by default. Authentication cookies are allowed when a site visitor hasn't consented to data collection. For more information, see <xref:security/gdpr#essential-cookies>.
 
-In `Startup.Configure`, call `UseAuthentication` and `UseAuthorization` to set the `HttpContext.User` property and run Authorization Middleware for requests. Call the `UseAuthentication` and `UseAuthorization` methods before calling `UseEndpoints`:
-
-[!code-csharp[](cookie/samples/3.x/CookieSample/Startup.cs?name=snippet2)]
-
 The <xref:Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationOptions> class is used to configure the authentication provider options.
 
-Set `CookieAuthenticationOptions` in the service configuration for authentication in the `Startup.ConfigureServices` method:
+ <xref:Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationOptions> can be set in the <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie*> method:
```
Comment (serpent5) [inline aspnetcore/security/authentication/cookie.md:38]:
```suggestion
Configure <xref:Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationOptions> in the <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie%2A> method:
```

File: aspnetcore/security/authentication/cookie.md [hunk 4]
```diff
@@ -4,53 +4,45 @@ author: rick-anderson
...
-        ...
-    });
-```
+[!code-csharp[](cookie/samples/6.x/CookieSample/Program.cs?name=snippet2&highlight=8-14)]
 
 ## Cookie Policy Middleware
 
-[Cookie Policy Middleware](xref:Microsoft.AspNetCore.CookiePolicy.CookiePolicyMiddleware) enables cookie policy capabilities. Adding the middleware to the app processing pipeline is order sensitive&mdash;it only affects downstream components registered in the pipeline.
+The 
+[Cookie Policy Middleware (GitHub Source)](https://github.com/dotnet/aspnetcore/blob/main/src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs) <xref:Microsoft.AspNetCore.Builder.CookiePolicyAppBuilderExtensions.UseCookiePolicy%2A> enables cookie policy capabilities. Middleware is processed in the order it's added.
```
Comment (serpent5) [inline aspnetcore/security/authentication/cookie.md:45]:
```suggestion
[Cookie Policy Middleware (GitHub Source)](https://github.com/dotnet/aspnetcore/blob/main/src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs) <xref:Microsoft.AspNetCore.Builder.CookiePolicyAppBuilderExtensions.UseCookiePolicy%2A> enables cookie policy capabilities. Middleware is processed in the order it's added:
```

File: aspnetcore/security/authentication/cookie.md [hunk 5]
```diff
@@ -232,32 +181,29 @@ await HttpContext.SignInAsync(
...
 
 ## Configuration
 
-If the app doesn't use the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app), create a package reference in the project file for the [Microsoft.AspNetCore.Authentication.Cookies](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Cookies/) package.
-
-In the `Startup.ConfigureServices` method, create the Authentication Middleware service with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication*> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie*> methods:
+In the `Startup.ConfigureServices` method, create the Authentication Middleware services with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication*> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie*> methods:
```
Comment (serpent5) [inline aspnetcore/security/authentication/cookie.md:194]:
```suggestion
In the `Startup.ConfigureServices` method, create the Authentication Middleware services with the <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication%2A> and <xref:Microsoft.Extensions.DependencyInjection.CookieExtensions.AddCookie%2A> methods:
```

**Review by serpent5 — APPROVED** [review]
Just a few teeny weeny suggestions.

**Pull request merged**